trin-core/src/utp/stream.rs

```diff
@@ -312,13 +320,22 @@ impl UtpListener {
                                 error!("Unable to send FIN packet to uTP stream handler");
                                 return;
                             }
-
                             let mut buf = [0; BUF_SIZE];
                             if let Err(msg) = conn.recv(&mut buf).await {
                                 error!("Unable to receive uTP FIN packet: {msg}")
                             }
+                        } else if let Some(conn) = self
+                            .utp_connections
+                            .get_mut(&ConnectionKey::new(*node_id, connection_id))
+                        {
+                            // Do not handle the packet here, send it to uTP socket layer
+                            if conn.discv5_tx.send(packet).is_err() {
+                                error!("Unable to send FIN packet to uTP stream handler");
+                            }
```

Comment:
I was surprised in this PR that the utp socket layer already handles this new case smoothly, but I'm assuming you tried it. Cool that the change ended up being pretty small!

Reply:
yeah, the uTP socket is designed in such a way to allow bidirectional packets, It is tested but we may want to add also some uTP tests in CI.